tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Steps/OdsOrganisationSteps.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using FluentAssertions;
+using IdentityModel.Client;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using NHSD.BuyingCatalogue.Identity.Api.IntegrationTests.Steps.Common;
+using NHSD.BuyingCatalogue.Identity.Api.IntegrationTests.Utils;
+using TechTalk.SpecFlow;
+using TechTalk.SpecFlow.Assist;
+
+namespace NHSD.BuyingCatalogue.Identity.Api.IntegrationTests.Steps
+{
+    [Binding]
+    public sealed class OdsOrganisationSteps
+    {
+        private readonly ScenarioContext _context;
+        private readonly Response _response;
+        private readonly OdsApiSteps _api;
+
+        private readonly string _organisationUrl;
+
+        public OdsOrganisationSteps(Response response, ScenarioContext context, Settings settings, OdsApiSteps api)
+        {
+            _response = response;
+            _context = context;
+            _api = api;
+            _organisationUrl = settings.OrganisationApiBaseUrl + "/api/v1/ods";
+        }
+
+        [Given(@"Ods Organisations exist")]
+        public async Task GivenOrganisationsExist(Table table)
+        {
+            var odsOrganisations = table.CreateSet<OdsApiResponseTable>();
+
+            foreach (OdsApiResponseTable org in odsOrganisations)
+            {
+                var odsApiOrganisation = TransformIntoOdsApiFormat(org);
+                var odsApiOrganisationAsJson = JsonConvert.SerializeObject(odsApiOrganisation);
+                await _api.SetUpGETEndpoint(org.OdsCode, odsApiOrganisationAsJson);
+            }
+        }
+
+        [When(@"a GET request is made for an Ods organisation with code (.*)")]
+        public async Task WhenAGETRequestIsMadeForAnOdsOrganisationWithOdsCode(string odsCode)
+        {
+            using var client = new HttpClient();
+            client.SetBearerToken(_context.Get(ScenarioContextKeys.AccessToken, ""));
+            _response.Result = await client.GetAsync(new Uri($"{_organisationUrl}/{odsCode}"));
+        }
+
+        [Then(@"the Ods Organisation is returned with the following values")]
+        public async Task ThenTheOrganisationIsReturnedWithTheFollowingValues(Table table)
+        {
+            var expectedOrganisation = table.CreateSet<OrganisationTable>().FirstOrDefault();
+
+            JToken responseBody = await _response.ReadBodyAsJsonAsync();
+
+            var actualOrganisation = CreateOrganisationFromJson(responseBody);
```

Comment:
Would using Flurl simplify this?

Reply:
With my limited knowledge I would lean towards saying no.

we could use its build-in serialization, however that would mean introducing yet another dto, (our 'Table' classes flatten out the 'Address' object so that they can be populated with specflow's tabular data with ease) and that might be slightly confusing to the reader.